tests/CMakeLists.txt

```diff
@@ -50,14 +54,16 @@ else()
   endforeach()
 endif()
 
-add_test(
-  NAME sources_format
-  COMMAND ${PYTHON_EXECUTABLE} run_sources_tests.py
-    --executable $<TARGET_FILE:uncrustify>
-    -d --git ${GIT_EXECUTABLE}
-    --result-dir ${CMAKE_CURRENT_BINARY_DIR}
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-)
+if (NOT ENABLE_CODECOVERAGE)
```

Comment:
Um... okay, I guess, but what's the problem this is fixing?

Reply:
I missed this:

> We should remove uncrustify itself from contributing to the coverage. Otherwise it will continue to report loss in coverage after code refactoring.

I'm not sure I understand why "it will continue to report loss in coverage after code refactoring". I *can* see an argument that we should not include coverage for what uncrustify does operating on its own sources, "because those aren't 'tests'", but OTOH they sort-of are, because we do have 'expected output' and raise an error if the expected and actual don't match.

Reply:
Imagine you're doing some refactoring and removing some code. That piece of code was triggering some other part from uncrustify and that was the only thing that was covering. Suddenly you're getting an unexpected drop in coverage and forced to add some coverage to a totally unrelated piece of uncrustify's code base.

With tests is different. There is a clear input/expected output and nothing should be changing underneath your feet while you are changing the code. The input/expected should remain static in this process.

Reply:
Okay, that makes sense. Thanks for the explanation!